runtime/lua/vim/lsp.lua

```diff
@@ -1015,5 +1015,21 @@ function lsp.get_log_path()
   return log.get_filename()
 end
 
+local function define_default_sign(name, properties)
+  if not vim.fn.sign_getdefined(name) then
+    vim.fn.sign_define(name, properties)
+  end
+end
+
+-- Define the LspDiagnostics signs if they're not defined already.
+local function define_default_lsp_diagnostics_signs()
+  define_default_sign('LspDiagnosticsErrorSign', {text='E', texthl='LspDiagnosticsError', linehl='', numhl=''})
+  define_default_sign('LspDiagnosticsWarningSign', {text='W', texthl='LspDiagnosticsWarning', linehl='', numhl=''})
+  define_default_sign('LspDiagnosticsInformationSign', {text='I', texthl='LspDiagnosticsInformation', linehl='', numhl=''})
+  define_default_sign('LspDiagnosticsHintSign', {text='H', texthl='LspDiagnosticsHint', linehl='', numhl=''})
+end
+
+define_default_lsp_diagnostics_signs()
```

Comment:
don't really need this, afaik idiomatic Lua uses `do...end` to create a new scope for something that's executed once.

```lua
do
  define_default_sign('LspDiagnosticsErrorSign', {text='E', texthl='LspDiagnosticsError', linehl='', numhl=''})
  define_default_sign('LspDiagnosticsWarningSign', {text='W', texthl='LspDiagnosticsWarning', linehl='', numhl=''})
  define_default_sign('LspDiagnosticsInformationSign', {text='I', texthl='LspDiagnosticsInformation', linehl='', numhl=''})
  define_default_sign('LspDiagnosticsHintSign', {text='H', texthl='LspDiagnosticsHint', linehl='', numhl=''})
end
```

Reply:
I think of the function name as a comment. It's pretty clear with `define_default_sign` what's going on but in case we add more, `lsp_init` could have hinted devs. @jakbyte you call. I can merge as is :)